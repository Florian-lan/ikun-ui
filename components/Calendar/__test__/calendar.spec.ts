import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import KCalendar from '../src';

let host;

const initHost = () => {
	host = globalThis.document.createElement('div');
	host.setAttribute('id', 'host');
	globalThis.document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.useRealTimers();
});

describe('Test: KCalendar', () => {
	test('props: cls', async () => {
		const instance = new KCalendar({
			target: host,
			props: {
				cls: 'k-calendar--test'
			}
		});
		expect(instance).toBeTruthy();
		expect(host!.innerHTML.includes('k-calendar--test')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
