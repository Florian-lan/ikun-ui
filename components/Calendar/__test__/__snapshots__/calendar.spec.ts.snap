// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KCalendar > props: cls 1`] = `"<div class="k-calendar k-calendar--base k-calendar--test"></div>"`;
